readthedocs/core/utils/__init__.py

```diff
@@ -125,6 +128,10 @@ def prepare_build(
     options['soft_time_limit'] = time_limit
     options['time_limit'] = int(time_limit * 1.2)
 
+    if build:
```

Comment:
Should this check for the `version` being external before sending?

Reply:
The status can be for any commit, I think it would be cool to report the status in normal versions

Reply:
> Should this check for the `version` being external before sending?

@ericholscher I'm checking that inside the task
https://github.com/readthedocs/readthedocs.org/blob/0011043e85c3499c66079893806b06d1065aea84/readthedocs/projects/tasks.py#L1807-L1808

We are using this in few places so it did not look good to check everywhere so I put it here.

Reply:
> The status can be for any commit, I think it would be cool to report the status in normal versions

We don't store the `commit sha` of a branch.

Reply:
> We are using this in few places so it did not look good to check everywhere so I put it here.

Yea, but it creates an entire task in order to be thrown away, which feels quite wasteful. I guess it's OK for now, but I'd really prefer to not have all those extra tasks triggered on normal builds.

Reply:
@ericholscher can we create a function that will check the version type and then call the task? or should we check for the type everywhere we use it?